public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -236,9 +270,66 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
 
   const styles = useStyles2(getStyles);
 
+  // Cloud alerts load data from form values
+  // whereas Grafana managed alerts load data from reducer
+  //when data source is changed in the cloud selector we need to update the queries in the reducer
```

Comment:
This sounds reasonable, but is there a reason we can't use the reducer state for both and always load the form state from the reducer state?

Reply:
I would prefer doing this change in a follow up PR as i think it's not a simple change. wdyt?